src/odemis/gui/cont/tabs.py

```diff
@@ -316,6 +316,516 @@ def get_display_priority(cls, main_data):
+            has_opt = any(isinstance(s, acqstream.OpticalStream)
+                          for s in self.tab_data_model.streams.value)
+            if not has_opt:
+                # We don't forbid to remove it, as for the user it can be easier
+                # to remove than change all the values
+                self._add_opt_stream(add_to_view=True, play=False)
+
+        if hasattr(self.tab_data_model, 'emState'):
+            has_sem = any(isinstance(s, acqstream.EMStream)
+                          for s in self.tab_data_model.streams.value)
+            if not has_sem:
+                stream_cont = self._add_em_stream(add_to_view=True, play=False)
+                stream_cont.stream_panel.show_remove_btn(False)
+
+    # TODO: move to stream controller?
+    # => we need to update the state of optical/sem when the streams are play/paused
+    # Listen to this event to just add (back) a stream if none is left when turning on?
+    def onOpticalState(self, state):
+        if state == guimod.STATE_ON:
+            # Pick the last optical stream that played (.streams is ordered)
+            for s in self.tab_data_model.streams.value:
+                if isinstance(s, acqstream.OpticalStream):
+                    opts = s
+                    break
+            else:  # Could happen if the user has deleted all the optical streams
+                sp = self._add_opt_stream(add_to_view=True)
+                opts = sp.stream
+
+            self._streambar_controller.resumeStreams({opts})
+            # focus the view
+            self.view_controller.focusViewWithStream(opts)
+        else:
+            self._streambar_controller.pauseStreams(acqstream.OpticalStream)
+
+    def onEMState(self, state):
+        if state == guimod.STATE_ON:
+            # Use the last SEM stream played
+            for s in self.tab_data_model.streams.value:
+                if isinstance(s, acqstream.EMStream):
+                    sems = s
+                    break
+            else:  # Could happen if the user has deleted all the EM streams
+                sp = self._add_em_stream(add_to_view=True)
+                sp.show_remove_btn(False)
+                sems = sp.stream
+
+            self._streambar_controller.resumeStreams({sems})
+            # focus the view
+            self.view_controller.focusViewWithStream(sems)
+        else:
+            self._streambar_controller.pauseStreams(acqstream.EMStream)
+
+    def Show(self, show=True):
+        assert (show != self.IsShown())  # we assume it's only called when changed
+        super(LocalizationTab, self).Show(show)
+
+        # pause streams when not displayed
+        if not show:
+            self._streambar_controller.pauseStreams()
+
+    @classmethod
+    def get_display_priority(cls, main_data):
+        # For SECOM/DELPHI and all simple microscopes
+        if main_data.role in ("secom", "delphi", "sem", "optical"):
+            return 2
```

Comment:
Where's `cryo-secom`? Also it's not apparent to me how's this tab is activated if this role is not checked.